### PR TITLE
User controller

### PR DESCRIPTION
**Pull request by soojison**
Now all the functions related to the user will be handled by users_controller instead of, like, 5 different controllers.

### CODE REVIEW

File: app/controllers/organizations_controller.rb [hunk 1]
```diff
@@ -41,7 +41,9 @@ def update
     #otherwise update the attributes
     if @organization.update_attributes(org_params)
       #handle successful update
-      redirect_to :root
+      byebug
```
Comment (m13hall) [inline app/controllers/organizations_controller.rb]: is this a debugger?

Comment (soojison) [inline app/controllers/organizations_controller.rb]: Oops, yes, forgot to remove that one

File: app/controllers/profile_controller.rb [hunk 2]
```diff
@@ -1,15 +1,10 @@
...
   def not_found
     fail ActionController::RoutingError.new('Profile Not Found')
   end
 
-  def edit
-    @interests = Interests.all
-  end
+  #def edit
+  #  @interests = Interests.all
+  #end
```
Comment (m13hall) [inline app/controllers/profile_controller.rb]: let's take out the commented-out things too

File: app/controllers/profile_controller.rb [hunk 3]
```diff
@@ -1,15 +1,7 @@
...
-    @skills = Skill.all
-  end
-
   def not_found
     fail ActionController::RoutingError.new('Profile Not Found')
   end
 
-  def edit
-    @interests = Interests.all
-  end
```
Comment (schmitzc) [inline app/controllers/profile_controller.rb]: ❓ Can you delete this controller now?

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -59,19 +61,44 @@ def create
...
+        end
       end
+      redirect_to @user #user_path
+      flash[:notice] = "Your account has been updated successfully."
+    else
+      redirect_to :root
     end
   end
 
+  def list
```
Comment (schmitzc) [inline app/controllers/users_controller.rb]: To follow Rails conventions, this also should be `index` instead of `list`

File: config/routes.rb [hunk 5]
```diff
@@ -18,71 +18,13 @@
...
-  get '/profile/user/'         => 'profile#show_user', as: 'user'
-  get '/users'                    => 'userlist#show'
-  #Testing searchkick through a stackoverflow suggestion:
-  #http://stackoverflow.com/questions/15459668/rails-search-form-submit-to-show-method
-  #routes.rb
-  get 'search_users', :to => 'userlist#show'
+  devise_scope :user do
+    get '/profile/user/:id'         => 'users#show', as: 'user'
+    get '/users'                    => 'users#list'
+    get 'search_users', :to         => 'users#list'
```
Comment (schmitzc) [inline config/routes.rb]: ❓ Why are there two routes for this?

**Comment by schmitzc** [comment]
👍 Much simpler now! It will be much easier for future teams of students to figure out.

**Pull request merged**